lib/matplotlib/cbook.py

```diff
@@ -361,9 +361,11 @@ class _BoundMethodProxy(object):
     Minor bugfixes by Michael Droettboom
     '''
     def __init__(self, cb):
+        self._hash = hash(cb)
+        self._callbacks = []
         try:
             try:
-                self.inst = ref(cb.im_self)
+                self.inst = ref(cb.im_self, self._destroy)
```

Comment:
I suspect that I am just being dense here.  I am worried about the circular reference that the `Proxy` objects holds a (hard) ref to the weakref and the weakref holds a (I assume) hard ref back to the object.  But I think that is ok as if the weakref dies the call-back will be be triggered and presumably released.  However in the case where we lose all refs to the `Proxy` object, there is the 3 element cycle (proxy.inst -> ref -> proxy._destroy).  But, again, I think that is the correct behaviour as if you register a call-back function you expect it to _always_ run and keeping the proxy object around in probably the right move.